include/flatbuffers/flatbuffers.h

```diff
@@ -697,6 +698,17 @@ class vector_downward {
     }
   }
 
+  void reset() {
+    assert(allocator_ != nullptr);
+    if (buf_ != nullptr) {
+      allocator_->deallocate(buf_, reserved_);
+    }
+    reserved_ = (initial_size_ + sizeof(largest_scalar_t) - 1) &
+                ~(sizeof(largest_scalar_t) - 1);
+    buf_ = allocator_->allocate(reserved_);
```

Comment:
There is still an allocation here.. As I understand it, in the single use case for this function, we reset this buffer and then throw it away.